### PR TITLE
Properly taking ownership of nccl subcomm (and destroying it)

### PR DESCRIPTION
**Pull request by cjnolet**

### CODE REVIEW

File: cpp/include/raft/comms/detail/std_comms.hpp [hunk 1]
```diff
@@ -116,6 +122,11 @@ class std_comms : public comms_iface {
   {
     requests_in_flight_.clear();
     free_requests_.clear();
+
+    if (own_nccl_comm_) {
+      ncclCommDestroy(nccl_comm_);
```
Comment (ChuckHastings) [inline cpp/include/raft/comms/detail/std_comms.hpp:127]: Should wrap in `RAFT_NCCL_TRY_NO_THROW`

**Review by ChuckHastings — COMMENTED** [review]
Thanks!  Will test in my branch before approving.

File: cpp/include/raft/comms/detail/std_comms.hpp (hunk 1 above)
Comment (cjnolet) [reply cpp/include/raft/comms/detail/std_comms.hpp]: Ah! Good catch.

**Comment by ChuckHastings** [comment]
This clears up the customer problem I was chasing, Thanks!

**Review by ChuckHastings — APPROVED** [review]

**Pull request merged**